### ISSUE TITLE
where global attention are used for hotpotqa

### ISSUE THREAD

**Issue by Fan-Luo** [allenai/longformer#141]
Hi

At top of the page 14 in the paper, you mentioned 

> "For Longformer, we use global attention to `input tokens` as well as `sentence and paragraph token`s." 

And in the issue #78, you explained the format of the context is this:

>  \<s> [question] q q q [/question] \<t> title 1 \</t> s_11 s_12 s_13 [/sent] s_21 s_22 s_23 [/sent] ... \<t> title 2 \</t> s_21 s_22 s_23 [/sent] .... [/par] yes no null \</s>

When you said `input tokens`, do you mean `<s> [question] q q q [/question] `? 
And for `sentence and paragraph tokens`, do you mean all  `[/sent]` and `<t> title </t>`?

Thank you

**Comment by armancohan** [comment]
Yes. For hotpotqa global attention is used in:
`<s> [question] q q q [/question]` as well as `<t>` and `[/sent]` tokens.

**Comment by Fan-Luo** [comment]
Thank you for the prompt reply.
As my understanding, global attention are not used on the tokens inside of title and `</t>`?
For example: `<t> Lisa Simpson </t>`, global attention used on `<t>`, but not `<Lisa Simpson </t>` ?
<br>

The paper also mentioned:
> extraction loss predicted from special tokens at the end of sentences and paragraphs.

For sentence, do you use logits computed from `sequence_output[:,sent_indexes,:]`, where `sent_indexes` corresponds to `[/sent]`?
For paragraphs, use logits computed from `sequence_output[:,para_indexes,:]`, would `para_indexes` corresponds to `<t>` or `</t>`?
I wonder how different would sequence_output of `<t>` and `<\t>` be , if global attention applied only to `<t>`, but no `</t>`?

Thank you

**Comment by armancohan** [comment]
Yes, that is correct. 

> I wonder how different would sequence_output of <t> and <\t> be , if global attention applied only to <t>, but no </t>?

For paragraphs, we applied global attention on the tokens that predicted the paragraph (`<t>`). We virtually saw no performance difference on using `<t>` or `</t>` for paragraph prediction global attention.

**Comment by Fan-Luo** [comment]
Thank you for your clarification.
